client/gutenberg/editor/utils.js

```diff
@@ -12,17 +12,33 @@ import debugFactory from 'debug';
 
 const debug = debugFactory( 'calypso:gutenberg' );
 
-export const overrideAPIPaths = siteSlug => {
-	// TODO: no API support for now. We'll also need to handle authorization here.
-	apiFetch.use( () => noop );
+export const applyAPIMiddlewares = siteSlug => {
+	//make authenticated calls using the WordPress.com REST Proxy
+	//bypassing the apiFetch call that uses window.fetch
+	//first middleware in, last out
+	apiFetch.use( options => {
+		return wpcomProxyRequest(
+			{
+				...options,
+				apiNamespace: 'wp/v2',
+			},
+			( error, body ) => {
+				if ( error ) {
+					return Promise.reject( error );
+				}
+
+				return Promise.resolve( body );
+			}
+		);
```

Comment:
apiFetch is supposed to return the response object if `parse=false` in the options. (We use the `window.fetch` response object shape as a public API), so if we want to be 100% compatible here, we need to support that case too. 

It's probably used on for `metaboxes` right now in Gutenberg and Metaboxes are not used in Calypso so it should work without it but that doesn't guarantee that there won't be any usage of the `parse` flag outside metaboxes in the future. (I think I saw a PR not merged yet using it)